src/shared/components/commit-label/CommitLabel.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import { Label, Tooltip } from '@patternfly/react-core';
+import { BitbucketIcon } from '@patternfly/react-icons/dist/js/icons/bitbucket-icon';
+import { GithubIcon } from '@patternfly/react-icons/dist/js/icons/github-icon';
+import { GitlabIcon } from '@patternfly/react-icons/dist/js/icons/gitlab-icon';
+import { GitProvider } from '../../utils/git-utils';
+
+const tipText = {
+  [GitProvider.GITHUB]: 'Open in GitHub',
+  [GitProvider.GITLAB]: 'Open in GitLab',
+  [GitProvider.BITBUCKET]: 'Open in BitBucket',
+};
+const providerIcon = {
+  [GitProvider.GITHUB]: <GithubIcon data-test-id="git-hub-icon" />,
+  [GitProvider.GITLAB]: <GitlabIcon data-test-id="git-lab-icon" />,
+  [GitProvider.BITBUCKET]: <BitbucketIcon data-test-id="bit-bucket-icon" />,
+};
+
+type CommitLabelProps = {
+  gitProvider: GitProvider | string;
+  sha: string;
+  shaURL: string;
+};
+const CommitLabel: React.FC<CommitLabelProps> = ({ gitProvider, sha, shaURL }) => {
+  const label = (
+    <Label
+      className="commit-label"
+      color="blue"
+      icon={providerIcon[gitProvider]}
+      isTruncated
```

Comment:
remove `isTruncated` because we are always showing 6 characters.

Reply:
well, then it _is_ truncated 😄